test/end-to-end/offer.js

```diff
@@ -1624,4 +1648,321 @@ describe("Offer endpoint tests", () => {
+                ownerName: test_company.name,
+            });
+
+            hidden_user_test_offer = await Offer.create({
+                ...generateTestOffer({ isHidden: true }),
+                owner: test_company._id.toString(),
+                ownerName: test_company.name,
+            });
+
+            await (new OfferService()).disable(hidden_user_test_offer._id, OfferConstants.HiddenOfferReasons.company);
+
+            disabled_test_offer = await Offer.create({
+                ...generateTestOffer({
+                    "publishDate": (new Date(Date.now() + (10 * DAY_TO_MS))).toISOString(),
+                    "publishEndDate": (new Date(Date.now() + (20 * DAY_TO_MS))).toISOString()
+                }),
+                owner: test_company._id.toString(),
+                ownerName: test_company.name,
+            });
+
+            await (new OfferService()).disable(disabled_test_offer._id, OfferConstants.HiddenOfferReasons.admin);
+        });
+
+        test("should fail if not logged in", async () => {
+            await test_agent
+                .del("/auth/login");
+
+            const res = await test_agent
+                .post(`/offers/${test_offer._id}/enable`)
+                .expect(HTTPStatus.UNAUTHORIZED);
+            expect(res.status).toBe(HTTPStatus.UNAUTHORIZED);
+            expect(res.body).toHaveProperty("error_code", ErrorTypes.FORBIDDEN);
+            expect(res.body).toHaveProperty("errors");
+            expect(res.body.errors).toContainEqual(ValidationReasons.INSUFFICIENT_PERMISSIONS);
+        });
+
+        test("should succeed if admin and offer hidden by default", async () => {
+            await test_agent
+                .post("/auth/login")
+                .send(test_user_admin)
+                .expect(HTTPStatus.OK);
+
+            const res = await test_agent
+                .post(`/offers/${hidden_default_test_offer._id}/enable`)
+                .expect(HTTPStatus.OK);
+            expect(res.body).toHaveProperty("isHidden", false);
+        });
+
+        test("should succeed if company and hidden by user/admin", async () => {
+            await test_agent
+                .post("/auth/login")
+                .send(test_user_company)
+                .expect(HTTPStatus.OK);
+
+            const res = await test_agent
+                .post(`/offers/${hidden_user_test_offer._id}/enable`)
+                .expect(HTTPStatus.OK);
+            expect(res.body).toHaveProperty("isHidden", false);
+        });
+
+        test("should fail if already shown offer by default", async () => {
+            const res = await test_agent
+                .post(`/offers/${test_offer._id}/enable`)
+                .expect(HTTPStatus.FORBIDDEN);
+            expect(res.body).toHaveProperty("error_code", ErrorTypes.FORBIDDEN);
+            expect(res.body).toHaveProperty("errors");
+            expect(res.body.errors).toContainEqual(ValidationReasons.OFFER_VISIBLE);
+        });
```

Comment:
This makes me wonder if this should be idempotent, so that if it is already visible, it wouldn't fail since the outcome is the same

Reply:
If so, change the http method to PUT